### DISCUSSION TITLE
strange error when reading parquet files

### DISCUSSION THREAD

**Discussion by qiangbo** [duckdb/duckdb#3347] [Q&A]
`import pandas as pd
import duckdb

qry =   """
        select
            count(distinct TRANSACTION_KEY)
        from
            'C:/data/**/2022*.parquet'
        """
data_out = duckdb.query(qry).fetchall()`

Error:

----> 7 data_out = duckdb.query(qry).fetchall()

RuntimeError: Failed to read Parquet file "C:\data\2022-01-02.parquet": column "55" in parquet file is of type VARCHAR, could not auto cast to expected type INTEGER for this column

**Comment by Alex-Monahan** [comment]
Hello!
Would you be able to provide the parquet files, or a representative parquet file that triggers the error? That would be a huge help!

**Comment by hannes** [comment]
I think its because the schema between the parquet files in the glob differs too much

**Reply by Mytherin** [reply]
To add to that, there's a good chance this error has been fixed by #3238

**Comment by qiangbo** [comment]
Thanks everybody's reply. 

I found the root cause of the issue: I am working with multiple parquet files and there is one column whose data type is string. There are a few parquet files ONLY have None values for that column. Below is code that can reproduce the error.

```
import pandas as pd
import duckdb

df_data1 = pd.DataFrame({"A":["abc","def"]})
df_data2 = pd.DataFrame({"A":[None,None]})
df_data1.to_parquet("test1.parquet")
df_data2.to_parquet("test2.parquet")

qry =   """
        select
            *
        from
            'test*.parquet'
        """
data_out = duckdb.query(qry).fetchall()




```

**Reply by hannes** [reply]
Is this still an issue with the latest master?
